tests/python_package_test/test_dask.py

```diff
@@ -60,9 +60,16 @@ def _create_ranking_data(n_samples=100, output='array', chunk_size=50, **kwargs)
     w = rnd.rand(X.shape[0]) * 0.01
     g_rle = np.array([len(list(grp)) for _, grp in groupby(g)])
 
-    if output == 'dataframe':
+    if output.startswith('dataframe'):
         # add target, weight, and group to DataFrame so that partitions abide by group boundaries.
         X_df = pd.DataFrame(X, columns=[f'feature_{i}' for i in range(X.shape[1])])
+        if output == 'dataframe-with-categorical':
+            cat_series = pd.Series(
+                np.random.choice(['a', 'b', 'y', 'z'], X.shape[0]),
+                dtype='category'
+            )
+            X_df['cat_col'] = cat_series
+            X = np.hstack((X, cat_series.cat.codes.values.reshape(-1, 1)))
         X = X_df.copy()
```

Comment:
I guess this line `X = np.hstack((X, cat_series.cat.codes.values.reshape(-1, 1)))` is useless because next line `X = X_df.copy()` overwrites `X`.

Reply:
ah, missed that! yes you're right